pkg/oc/cli/admin/migrate/scc/scc.go

```diff
@@ -0,0 +1,357 @@
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(name, f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			kcmdutil.CheckErr(options.Run(cmd, out))
+		},
+	}
+	options.ResourceOptions.Bind(cmd)
+
+	// these flags aren't applied for SCC migrations
+	cmd.Flags().MarkHidden("to-key")
+	cmd.Flags().MarkHidden("from-key")
+	cmd.Flags().MarkHidden("include")
+	cmd.Flags().MarkHidden("filename")
+	cmd.Flags().MarkHidden("show-all")
+	cmd.Flags().MarkHidden("no-headers")
+	cmd.Flags().MarkHidden("show-labels")
+	cmd.Flags().MarkHidden("all-namespaces")
+
+	return cmd
+}
+
+func (o *migrateSCCOptions) Complete(name string, f kcmdutil.Factory, c *cobra.Command, args []string) error {
+	o.ResourceOptions.SaveFn = o.save
+	if err := o.ResourceOptions.Complete(f, c); err != nil {
+		return err
+	}
+	kubeClient, err := f.KubernetesClientSet()
+	if err != nil {
+		return err
+	}
+	o.kubeClient = kubeClient
+	o.cachedClusterRoles, err = o.kubeClient.RbacV1().ClusterRoles().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.cachedRoles, err = o.kubeClient.RbacV1().Roles(metav1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *migrateSCCOptions) Validate() error {
+	if o.kubeClient == nil {
+		return errors.New("Kubernetes client is required")
+	}
+	if len(o.ResourceOptions.Include) != 1 || o.ResourceOptions.Include[0] != "securitycontextconstraints.security.openshift.io" {
+		return fmt.Errorf("the only supported resources are securitycontextconstraints")
+	}
+	return o.ResourceOptions.Validate()
+}
+
+func (o *migrateSCCOptions) Run(cmd *cobra.Command, out io.Writer) error {
+	return o.ResourceOptions.Visitor().Visit(func(info *resource.Info) (migrate.Reporter, error) {
+		_, wasSCC := info.Object.(*securityapi.SecurityContextConstraints)
+		if !wasSCC {
+			return nil, fmt.Errorf("unrecognized object %#v", info.Object)
+		}
+
+		// all SCCs have to be migrated
+		return migrate.ReporterBool(true), nil
+	})
+}
+
+// save invokes the API to alter an object. The reporter passed to this method is the same returned by
+// the migration visitor method. It should return an error  if the input type cannot be saved
+// It returns migrate.ErrRecalculate if migration should be re-run on the provided object.
+func (o *migrateSCCOptions) save(info *resource.Info, reporter migrate.Reporter) error {
+	scc, wasSCC := info.Object.(*securityapi.SecurityContextConstraints)
+	if !wasSCC {
+		return fmt.Errorf("unrecognized object %#v", info.Object)
+	}
+
+	// TODO: do we need always return migrate.DefaultRetriable(info, err) ?
+
+	psp, err := convertSccToPsp(scc)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: could not create PSP for SCC %q: %v\n", scc.Name, err)
+		return err
+	}
+
+	role, err := convertSccToClusterRole(scc)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: could not create cluster role for SCC %q: %v\n", scc.Name, err)
```

Comment:
Every command should accept `genericclioptions.IOStreams` and use those as input/output. Never directly print to `os.Stderr`.

Reply:
I'll probably create a new PR for the `migrate` module for reasons from this comment https://github.com/openshift/origin/pull/18885#discussion_r207219690